src/backend/api/routes/api/get_commits_per_day.route.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/python3
+# -*- coding=utf-8 -*-
+r"""
+
+"""
+import fastapi
+import pydantic
+from api.database.models import Repository, Commit
+from api.database.db import createLocalSession
+from sqlalchemy import select, func
+from sqlalchemy.sql import extract
+
+router = fastapi.APIRouter()
+
+class ResponseModel(pydantic.BaseModel):
+    commits_per_day: dict[str, int]
+
+    class Config:
+        orm_mode = True
+
+@router.get("/commits_per_day/{workspace_id}/{year}", response_model=ResponseModel)
+async def get_commits_per_day(workspace_id: int, year: int):
+    r"""
+    get commits per day in the workspace from a specific year
+    """
+    with createLocalSession() as connection:
+        result = connection.execute(
+            select(func.count(), func.date(Commit.committed_at))
+            .group_by(func.date(Commit.committed_at))
+            .join(Repository.commits)
+            .filter(Repository.workspace_id == workspace_id)
+            .filter(extract('year', Commit.committed_at) == year)
+        ).all()
```

Comment:
vllt das? mindestens das da drüber als connection.query() schreiben. vllt leicht anpassen (der for-loop darunter stört mich sehr)
```py
import sqlalchemy as sql
import datetime as dt
from database import createLocalSession, models as dbm


with createLocalSession() as session:
    session.query(sql.func.count(), sql.func.date(dbm.Commit.committed_at))\
        .join(dbm.Repository)\
        .filter(dbm.Repository.workspace_id == workspace_id,
                sql.between(dbm.Commit.committed_at, dt.date(year=year), dt.date(year + 1)))\
        .group_by(sql.func.year(dbm.Commit.committed_at),
                  sql.func.month(dbm.Commit.committed_at),
                  sql.func.day(dbm.Commit.committed_at))\
        .all()
```

Reply:
jetzt wird hier auch .query verwendet. das group_by year und month ist unnötig, oder? es wird ja eh nur ein bestimmtes jahr zurückgegeben und wozu brauchen wir dann noch die monate?